src/core/Akka/Actor/ActorCell.cs

```diff
@@ -331,30 +359,33 @@ private static long NewUid()
         private ActorBase NewActor()
         {
             PrepareForNewActor();
-            ActorBase instance = null;
+            ActorBase? instance = null;
             //set the thread static context or things will break
             UseThreadContext(() =>
             {
                 _state = _state.ClearBehaviorStack();
                 instance = CreateNewActorInstance();
                 //TODO: this overwrites any already initialized supervisor strategy
                 //We should investigate what we can do to handle this better
-                instance.SupervisorStrategyInternal = _props.SupervisorStrategy;
+                instance.SupervisorStrategyInternal = Props.SupervisorStrategy;
                 //defaults to null - won't affect lazy instantiation unless explicitly set in props
             });
-            return instance;
+
+
+            Assert.Assert(instance != null, (string)(nameof(instance) + " != null"));
+            return instance!;
```

Comment:
The assertion on the line above prevents this from ever actually returning `null`.